package-build.el

```diff
@@ -828,7 +838,14 @@ in `package-build-archive-dir'."
           (package-build--copy-package-files files source-dir target)
           (package-build--write-pkg-file desc target)
           (package-build--generate-info-files files source-dir target)
-          (package-build--create-tar name version tmp-dir)
+          (let ((tstamp (replace-regexp-in-string
+                         (concat "^\\([0-9]{4}\\)\\([0-9]{2}\\)\\([0-9]{2}\\)"
+                                 "\\.\\([0-9]{2}\\)\\([0-9]{2}\\)")
+                         "\\1-\\2-\\3T\\4:\\5Z"
+                         (package-build--parse-time
+                          (package-build--get-timestamp rcp commit)
+                          (oref rcp time-regexp)))))
+            (package-build--create-tar name version tmp-dir tstamp))
```

Comment:
I would prefer doing all the work inside `package-build--create-tar`.

Reply:
But currently `package-build--create-tar` doesn't have the means to do so. Would you prefer if `rcp` and `commit` are passed as parameters? I found this to be the less obvious solution in comparison to the tstamp parameter that has a clear rationale.

Reply:
Makes sense. I guess what really bothers me is how much code this requires and I would rather hide it away.

Better yet, make it unnecessary. I have done that now by removing some existing complexity first. I'll keep the call to `package-build--get-timestamp` (approximately) where it is now, the rest isn't necessary anymore.

I'll have to test the refactoring a bit before pushing that and will likely do that tomorrow.